pkg/auto/exporthttp/job/job.go

```diff
@@ -48,50 +47,30 @@ type Job interface {
 
 type sender func(ctx context.Context, url string, body []byte) error
 
-// Mulpx for multiple Job easy chan fan-in
-type Mulpx struct {
-	C     chan Job
-	group *errgroup.Group
-}
-
-func (m *Mulpx) WaitForDone() {
-	// no errors are returned from any of the group.Go() calls
-	_ = m.group.Wait()
-	close(m.C)
-}
-
-// Multiplex receives multiple jobs and fans all the jobs into a single chan
-func Multiplex(ctx context.Context, jobs ...Job) *Mulpx {
-	group, ctx := errgroup.WithContext(ctx)
-
-	out := &Mulpx{
-		C:     make(chan Job, len(jobs)),
-		group: group,
-	}
+// ExecuteAll receives multiple jobs and executes all of them in an errgroup concurrently
+func ExecuteAll(ctx context.Context, sender sender, jobs ...Job) *errgroup.Group {
```

Comment:
I think your func api will be clearer if this just returned an error (and called group.Wait itself).